reviewqueue/helpers.py

```diff
@@ -140,7 +141,27 @@ def get_charmstore_entity(
     return charmstore._meta(entity_id, includes, channel=channel)
 
 
-def charm_pull(charmstore_url, dest_dir):
+class CharmFetchError(RuntimeError):
+    def __init__(self, message, cmd):
+        super(CharmFetchError, self).__init__(message)
+        self.cmd = cmd
+        self.hint = self._hint(message)
+
+    def _hint(self, msg):
+        """Parse the original error message and attempt to translate it
+        into something user-friendly -- a message that can guide the user
+        towards a solution.
+
+        """
+        if re.search(r"user is not a member of required groups", msg):
+            return (
+                "Ensure that all terms for this charm have been released "
+                "using the `charm release-term` command."
```

Comment:
I'm ignorant on when you'd get `user is not a member`, but is it always the case that `release-term` is the right hint?  Feels like you might also get this if the charm perms didn't give everyone 'read' perms.